cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -518,6 +518,7 @@ public void onNamespaceCreation(String cdapNamespace, Map<String, String> proper
     // and must start and end with an alphanumeric character
     KubeUtil.validateRFC1123LabelName(namespace);
     findOrCreateKubeNamespace(namespace, cdapNamespace);
+    twillRunner.addAndStartJobWatcher(cdapNamespace);
```

Comment:
it would be better to do this at the very end. Otherwise if any of the below operations fail, we'll have an extra monitoring thread.